go/feast/registry.go

```diff
@@ -0,0 +1,277 @@
+package feast
+
+import (
+	"errors"
+	"fmt"
+	"github.com/feast-dev/feast/go/protos/feast/core"
+	"github.com/golang/protobuf/proto"
+	"io/ioutil"
+)
+
+/*
+	Store protos of FeatureView, FeatureService, Entity,
+		OnDemandFeatureView, RequestFeatureView
+	but return to user copies of non-proto versions of these objects
+*/
+
+type Registry struct {
+	path                       string
+	cachedFeatureServices      map[string]map[string]*core.FeatureService
+	cachedEntities             map[string]map[string]*core.Entity
+	cachedFeatureViews         map[string]map[string]*core.FeatureView
+	cachedOnDemandFeatureViews map[string]map[string]*core.OnDemandFeatureView
+	cachedRequestFeatureViews  map[string]map[string]*core.RequestFeatureView
+}
+
+func NewRegistry(path string) (*Registry, error) {
+	// Read the local registry
+	in, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	registry := &core.Registry{}
+	if err := proto.Unmarshal(in, registry); err != nil {
+		return nil, err
+	}
+	r := &Registry{path: path,
+		cachedFeatureServices:      make(map[string]map[string]*core.FeatureService),
+		cachedEntities:             make(map[string]map[string]*core.Entity),
+		cachedFeatureViews:         make(map[string]map[string]*core.FeatureView),
+		cachedOnDemandFeatureViews: make(map[string]map[string]*core.OnDemandFeatureView),
+		cachedRequestFeatureViews:  make(map[string]map[string]*core.RequestFeatureView),
+	}
+	loadEntitiesDone := make(chan struct{}, 1)
+	loadFeatureServicesDone := make(chan struct{}, 1)
+	loadFeatureViewsDone := make(chan struct{}, 1)
+	loadOnDemandFeatureViewsDone := make(chan struct{}, 1)
+	loadRequestFeatureViewsDone := make(chan struct{}, 1)
+	doneLoad := make(chan struct{}, 1)
+	go func() {
```

Comment:
What is the performance benefit of this go routine vs serial execution?

Reply:
I haven't done a benchmarking on this nor added this to my current benchmarking code since it doesn't affect get_online_features() performance directly. But the idea is we can load lists of feature views, entities, feature services, on-demand feature views, etc from the registry in parallel. This might be beneficial if we add a refresh function to reload the registry cache multiple times?

Reply:
It seems like a premature optimization to me

Reply:
I will remove the goroutines